src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -211,14 +239,28 @@ impl LogicalAgg {
                                 .collect(),
                             _ => vec![],
                         };
-                        let state = get_merialized_input_state(sort_keys, include_keys);
-                        AggCallState::MaterializedInputState(Box::new(state))
+                        let (table, column_mapping) =
+                            gen_state_table(sort_keys, include_keys, true);
+                        let state = MaterializedAggInputState {
+                            table,
+                            column_mapping,
+                        };
+                        AggCallState::MaterializedInput(Box::new(state))
                     } else {
-                        AggCallState::ResultValueState
+                        AggCallState::ResultValue
                     }
                 }
                 AggKind::Sum | AggKind::Count | AggKind::Avg | AggKind::ApproxCountDistinct => {
-                    AggCallState::ResultValueState
+                    AggCallState::ResultValue
+                }
+                AggKind::SinglePhaseAppendOnlyApproxDistinct => {
+                    if !in_append_only {
+                        panic!("SinglePhaseAppendOnlyApproxDistinct can only be used in append-only stream.");
+                    } else {
+                        let (table, _) = gen_state_table(vec![], vec![], false);
```

Comment:
I would suggest generating state table for this agg kind with a separate function, because the logic is totally different.

Reply:
> I would suggest generating state table for this agg kind with a separate function, because the logic is totally different.

fixed